pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -527,37 +553,35 @@ func (r *SubnetPortReconciler) addressBindingMapFunc(ctx context.Context, obj cl
 		log.Info("Invalid object", "type", reflect.TypeOf(obj))
 		return nil
 	}
-	// skip reconcile if AddressBinding exists and is realized
-	if ab.Status.IPAddress != "" {
-		namespacedName := types.NamespacedName{
-			Name:      ab.Name,
-			Namespace: ab.Namespace,
-		}
-		existingAddressBinding := &v1alpha1.AddressBinding{}
-		if err := r.Client.Get(context.TODO(), namespacedName, existingAddressBinding); err == nil {
-			return nil
-		}
-	}
 	spList := &v1alpha1.SubnetPortList{}
 	spIndexValue := fmt.Sprintf("%s/%s", ab.Namespace, ab.Spec.VMName)
 	err := r.Client.List(context.TODO(), spList, client.MatchingFields{util.SubnetPortNamespaceVMIndexKey: spIndexValue})
-	if err != nil || len(spList.Items) == 0 {
+	if err != nil {
 		log.Error(err, "Failed to list SubnetPort from cache", "indexValue", spIndexValue)
 		return nil
 	}
+	if len(spList.Items) == 0 {
+		setAddressBindingStatus(r.Client, ctx, ab, metav1.Now(), vmOrInterfaceNotFoundError, "")
+		return nil
+	}
+	// sort by CreationTimestamp
+	slices.SortFunc(spList.Items, func(a, b v1alpha1.SubnetPort) int {
+		return a.CreationTimestamp.UTC().Compare(b.CreationTimestamp.UTC())
+	})
 	if ab.Spec.InterfaceName == "" {
 		if len(spList.Items) == 1 {
 			log.V(1).Info("Enqueue SubnetPort for default AddressBinding", "namespace", ab.Namespace, "name", ab.Name, "SubnetPortName", spList.Items[0].Name, "VM", ab.Spec.VMName)
-			return []reconcile.Request{{
-				NamespacedName: types.NamespacedName{
-					Name:      spList.Items[0].Name,
-					Namespace: spList.Items[0].Namespace,
-				},
-			}}
 		} else {
-			log.Info("Found multiple SubnetPorts for a VM, ignore default AddressBinding for SubnetPort", "namespace", ab.Namespace, "name", ab.Name, "subnetPortCount", len(spList.Items), "VM", ab.Spec.VMName)
-			return nil
-		}
+			// Reconcile the oldest SubnetPort to check if the ExternalAddress should be removed.
+			log.Info("Found multiple SubnetPorts for a VM, enqueue oldest SubnetPort", "namespace", ab.Namespace, "name", ab.Name, "subnetPortCount", len(spList.Items), "VM", ab.Spec.VMName)
```

Comment:
I do not get this, why we need to reconcile the first subnetport if multiple subnetports found ?

Reply:
For below case:
1. Crate SubnetPort1 for vm1/inf1
2. Create AddressBinding for vm1 (inf is not provided)
3. Create SubnetPort2 for vm1/inf2
To avoid unexpected behavior, we'd better enqueue SubnetPort1 to remove the external IP.

Reply:
still confused, if vm1 has 2 interfaces, then AddressBinding1 without interface will be "MultipleInterface" error? it cannot have the external IP.